swig/cpp/src/Tree.h

```diff
@@ -0,0 +1,99 @@
+/**
+ * @file Trees.h
+ * @author Mislav Novakovic <mislav.novakovic@sartura.hr>
+ * @brief Sysrepo class header for C header trees.h.
+ *
+ * @copyright
+ * Copyright 2016 Deutsche Telekom AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef TREE_H
+#define TREE_H
+
+#include "Sysrepo.h"
+#include "Struct.h"
+
+extern "C" {
+#include "sysrepo.h"
+#include "sysrepo/trees.h"
+}
+
+using namespace std;
+
+class Tree:public Throw_Exception
+{
+public:
+    Tree(const char *root_name, const char *root_module_name);
+    Tree(sr_node_t *tree, bool free = false);
+    shared_ptr<Tree> dup();
+    shared_ptr<Tree> node();
+    sr_node_t *tree() {return _node;};
+    sr_mem_ctx_t *_sr_mem() {return _node->_sr_mem;};
```

Comment:
_sr_mem is meant to be and described as an internal member already in the C API so I wouldn't make it accessible through the C++ API.